### ISSUE TITLE
Predict without censoring

### ISSUE THREAD

**Issue by athowes** [epinowcast/epidist#256]
While working on #52 I've come across this bug. I'd like to do prediction of the delay without any censoring. I think that this can be achieved by changing the values of `pwindow_upr` and `swindow_upr` passed as `newdata`. (I don't like these names and think they would be better as `pwindow` and `swindow` or `pwindow_width` and `swindow_width`, as I have written in https://github.com/epinowcast/epidist/issues/245).

Doing this:

```r
newdata_pdf <- obs_prep |>
  as.data.frame() |>
  modelr::data_grid(NA) |>
  dplyr::mutate(obs_t = 1000, pwindow_upr = 0, swindow_upr = 0)

draws_pdf <- predicted_draws(fit, newdata = newdata_pdf, ndraws = 1)
```

```
Error in while (d_censored > obs_t) { : 
  missing value where TRUE/FALSE needed
```

To demonstrate that this is a reasonable approach, if I were to set `pwindow_upr` and `swindow_upr` to 0.01 then you get a draw up to a precision of 0.01:

```
# Groups:   NA, obs_t, pwindow_upr, swindow_upr, .row [1]
  `NA`  obs_t pwindow_upr swindow_upr  .row .chain .iteration .draw .prediction
  <lgl> <dbl>       <dbl>       <dbl> <int>  <int>      <int> <int>       <dbl>
1 NA     1000        0.01        0.01     1     NA         NA     1        2.87
```

Of course one could just set these numbers to be 0.0000001 etc. but that feels hacky. I might prefer to see if there is something that could be done within the prediction function to work around this.

**Comment by athowes** [comment]
Seems pretty clear that there is something that can be done in the special case of 0 here:

```r

  obs_t <- prep$data$vreal1[i]
  pwindow_width <- prep$data$vreal2[i]
  swindow_width <- prep$data$vreal3[i]

  # while loop to impose the truncation
  d_censored <- obs_t + 1
  while (d_censored > obs_t) {
    p_latent <- runif(1, 0, 1) * pwindow_width
    d_latent <- rlnorm(1, meanlog = mu, sdlog = sigma)
    s_latent <- p_latent + d_latent
    p_censored <- floor_mult(p_latent, pwindow_width)
    s_censored <- floor_mult(s_latent, swindow_width)
    d_censored <- s_censored - p_censored
  }
 ```

**Comment by athowes** [comment]
I think this should be an easy fix but I'm leaving it until after #261 is merged as it's building on that same function.
